cedar-policy-core/src/est/expr.rs

```diff
@@ -735,7 +721,13 @@ impl From<ast::Expr> for Expr {
             ast::ExprKind::Lit(lit) => lit.into(),
             ast::ExprKind::Var(var) => var.into(),
             ast::ExprKind::Slot(slot) => slot.into(),
-            ast::ExprKind::Unknown(ast::Unknown { name, .. }) => Expr::unknown(name),
+            ast::ExprKind::Unknown(ast::Unknown { name, .. }) => {
+                // Create an extension call with one arg, which is the name of the unknown
+                Expr::ext_call(
+                    "unknown".to_string().into(),
+                    vec![Expr::lit(CedarValueJson::String(name))],
+                )
+            }
```

Comment:
continuing my earlier comment, then we wouldn't need this change at all; this could still be a call to `Expr::unknown`